File: engine/model/src/main/scala/net/server/channel/packet/movement/MovePlayerPacket.scala
package net.server.channel.packet.movement

import net.server.{MaplePacket, MovementData}

class MovePlayerPacket( private var _hasMovement: Boolean,  private var _movementDataList: java.util.List[MovementData],  private var _movementList: java.util.List[java.lang.Byte]) extends BaseMovementPacket {
     def hasMovement: Boolean = _hasMovement
     def movementDataList: java.util.List[MovementData] = _movementDataList
     def movementList: java.util.List[java.lang.Byte] = _movementList
}

File: engine/engine-model/src/main/java/net/server/channel/packet/movement/MovePlayerPacket.java
package net.server.channel.packet.movement;

import java.util.List;

import net.server.MovementData;

public class MovePlayerPacket extends BaseMovementPacket {
   private final Boolean hasMovement;

   private final List<MovementData> movementDataList;

   private final List<Byte> movementList;

   public MovePlayerPacket(Boolean hasMovement, List<MovementData> movementDataList, List<Byte> movementList) {
      this.hasMovement = hasMovement;
      this.movementDataList = movementDataList;
      this.movementList = movementList;
   }

   public Boolean hasMovement() {
      return hasMovement;
   }

   public List<MovementData> movementDataList() {
      return movementDataList;
   }

   public List<Byte> movementList() {
      return movementList;
   }
}
